File: internal/shared/time.go
package shared

import (
	"time"
)

var jst = time.FixedZone("JST", 9*60*60)

func ToJSTDateString(t time.Time) string {
	return t.In(jst).Format("2006-01-02")
}

func AgeHours(now time.Time, publishedAt string) float64 {
	pt, err := time.Parse(time.RFC3339, publishedAt)
	if err != nil {
		return 999
	}
	d := now.Sub(pt)
	if d < 0 {
		return 0
	}
	return d.Hours()
}

func ShouldPostOnDate(jstDate string) bool {
	anchor, _ := time.ParseInLocation("2006-01-02", "2026-03-02", jst)
	current, err := time.ParseInLocation("2006-01-02", jstDate, jst)
	if err != nil {
		return false
	}
	days := int(current.Sub(anchor).Hours() / 24)
	return days%2 == 0
}

File: src/shared/time.ts
import { ANCHOR_JST_DATE, JST_TIMEZONE } from "../domain/model.js";

export function toJstDateString(date: Date): string {
  const fmt = new Intl.DateTimeFormat("en-CA", {
    timeZone: JST_TIMEZONE,
    year: "numeric",
    month: "2-digit",
    day: "2-digit"
  });
  return fmt.format(date);
}

export function ageHours(now: Date, publishedAt: string): number {
  const t = new Date(publishedAt).getTime();
  if (Number.isNaN(t)) {
    return 999;
  }
  const diffMs = Math.max(0, now.getTime() - t);
  return diffMs / (1000 * 60 * 60);
}

export function shouldPostOnDate(jstDate: string): boolean {
  const anchor = Date.parse(`${ANCHOR_JST_DATE}T00:00:00+09:00`);
  const current = Date.parse(`${jstDate}T00:00:00+09:00`);
  const days = Math.floor((current - anchor) / (1000 * 60 * 60 * 24));
  return days % 2 === 0;
}
